Sources/NIOCore/ChannelPipeline.swift

```diff
@@ -457,10 +457,10 @@ public final class ChannelPipeline: ChannelInvoker {
         let promise = self.eventLoop.makePromise(of: ChannelHandlerContext.self)
 
         if self.eventLoop.inEventLoop {
-            promise.completeWith(self.contextSync(handler: handler))
+            promise.assumeIsolated().completeWith(self.contextSync(handler: handler))
```

Comment:
`assumeIsolated()` is a noop in `release`, right?

Reply:
Right now it is but it needs to change. It uses `assert`s to check but we need to upgrade this `preconditions` since check at creation time is not enough similar to loop bound. Here is an example where just a `precondition` on creation of the `assumeIsolated` would fail

```
// On the EL
let assumeIsolated = someFuture.assumeIsolated()
assumeIsolated.wait() // Fine without a compiler error since it is no longer required to be Sendable. Also no runtime crash

await withTaskExecutorPreference(someRandomExec) {
  assumeIsolated.wait() // No compiler error but runtime crash now
}
```

Reply:
@FranzBusch the line above is `if self.eventLoop.inEventLoop()`, so the `assumeIsolated()` is duplicated.

Reply:
Ah sorry, I thought you were asking in a more general term. In this particular case I don't know if the compiler is able to optimise the precondition away.

Reply:
No, it doesn't do that sort of thing. If we're not sure that this is a noop, we should remove it here. Already not super cheap to check it

Reply:
Or do `assertIsolated` not `preconditionIsolated`